sunbeamlib/__init__.py

```diff
@@ -3,9 +3,17 @@
 
 from snakemake.utils import listfiles
 from snakemake.workflow import expand
+import os
 
 
 def build_sample_list(data_fp, filename_fmt, excluded):
+    if os.path.isdir(str(data_fp)):
+        Samples = build_sample_from_dir(data_fp, filename_fmt, excluded)
+    else:
+        Samples = build_sample_from_file(data_fp)
+    return Samples
+
+def build_sample_from_dir(data_fp, filename_fmt, excluded):
```

Comment:
let's make these sub-functions start with an underscore to emphasize that they're not supposed to be called directly, and pluralize samples: _build_samples_from_dir and _build_samples_from_file